code/modules/mining/drilling/drill.dm

```diff
@@ -51,10 +51,14 @@
 /obj/machinery/mining/drill/Destroy()
 	for(var/obj/machinery/mining/brace/b in supports)
 		b.disconnect()
+	qdel(soul)
+	soul = null
```

Comment:
You can replace this with QDEL_NULL(soul), same effect

Reply:
thats spooky way of doing it

Reply:
its a cleaner way